File: 347.top-k-frequent-elements.cpp
/*
 * @lc app=leetcode id=347 lang=cpp
 *
 * [347] Top K Frequent Elements
 */
#include <bits/stdc++.h>

using namespace std;

// @lc code=start
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> hash_table;
        for (int num: nums) {
            hash_table[num]++;
        }
        vector<pair<int, int>> temp_vector(hash_table.begin(), hash_table.end());
        sort(temp_vector.begin(), temp_vector.end(), [](pair<int, int>a, pair<int, int>b) {
            return a.second > b.second;
        });
        vector<int> result;
        for (int i=0; i<k; i++) {
            result.push_back(temp_vector[i].first);
        }
        return result;
    }
};
// @lc code=end

File: 347.top-k-frequent-elements.py
#
# @lc app=leetcode id=347 lang=python3
#
# [347] Top K Frequent Elements
#
from collections import Counter
from typing import List

# @lc code=start
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        return [n for n, _ in Counter(nums).most_common(k)]
# @lc code=end

